src/Datasets.php

```diff
@@ -42,12 +42,69 @@ public static function set(string $name, $data): void
     public static function get(string $name)
     {
         if (!array_key_exists($name, self::$datasets)) {
-            throw new DatasetDoesNotExist($name);
+            return self::extractDatasetKeys($name);
         }
 
         return self::$datasets[$name];
     }
 
+    public static function unset(string $name): void
+    {
+        if (array_key_exists($name, self::$datasets)) {
+            unset(self::$datasets[$name]);
+        }
+    }
+
+    /**
+     * @return array[]
+     */
+    private static function extractDatasetKeys(string $datasetName): array
+    {
+        $originalDatasetName = $datasetName;
+
+        $pieces = explode(':', $datasetName);
+
+        if ($pieces == false) {
+            throw new DatasetDoesNotExist($originalDatasetName);
+        }
+
+        $datasetName = trim($pieces[0]);
```

Comment:
By taking the first element from the exploded array, we actually disallow this feature to be used together with dataset that has `:` in them.
So we could change this to say that the dataset name is everything before the last `:` instead.

Meaning if I have a dataset named `My:dataset:name,age`, it will see the dataset name as `My:dataset` and the keys as `name,age`

Can be done by replacing the `explode` with a simple regex like this
`\^(.*):(.*)\`
So your first capture group is the dataset name and the second is your keys 👍 (possibly also other solutions)

Reply:
done!